app/services/ignore_file_service.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+# Service for determining if a file provided by a user should be ignored.
+class IgnoreFileService
+  # @return [Boolean] true if the file should be ignored, false otherwise
+  def self.call(filepath:)
```

Comment:
Borrowed from H2.

Reply:
Does this address the example in Amy's ticket of a temporary file with a name starting with `~$`?

Reply:
It does now. It turns out these are MS Office temporary files and were not ignored in H2.